misk/src/test/kotlin/org/assertj/core/api/AssertExtensions.kt

```diff
@@ -19,4 +20,38 @@ fun <ACTUAL : CharSequence> AbstractCharSequenceAssert<*, ACTUAL>.isEqualToAsJso
     return this
 }
 
+class OrderingAssert<in A : Comparable<A>>(private val actual: A) {
+    infix fun comesBefore(other: A) {
+        assertThat(actual).isLessThan(other)
+        assertThat(other).isGreaterThan(actual)
+    }
 
+    infix fun comesAfter(other: A) {
+        assertThat(actual).isGreaterThan(other)
+        assertThat(other).isLessThan(actual)
+    }
+
+    infix fun eq(other: A) {
+        assertThat(actual).isEqualByComparingTo(other)
+    }
+}
+
+fun <A : Comparable<A>> assertOrdering(vararg values: A) {
+    values.forEachIndexed { index, value ->
+        assertThat(value).isEqualByComparingTo(value)
+
+        val before = if (index == 0) listOf() else values.take(index - 1)
+        before.forEach {
+            assertThat(value).isGreaterThan(it)
```

Comment:
this assumes asc ordering right? maybe name the method `assertAscOrdering`

Reply:
also why not just use the built in sorted asserts? `assertThat(values).isSorted`

Reply:
really hard to debug when the sorting goes wrong

Reply:
yea that's fair. maybe just add a note, so a future contributor doesn't get smart and change it